nuxeo-justified-grid/nuxeo-justified-grid.html

```diff
@@ -287,14 +336,111 @@
+       * they fit. When a row doesn't have space to fit another item, we increase the height of the current row to take
+       * advantage of the remaining width space to avoid cropping images.
+       * Currently relies on `picture:info` to get the dimensions of a video/image. Fallbacks to square dimensions if the
+       * document doesn't have the `picture` schema.
+       */
+      _computeRows: function(items) {
+        var gridWidth = this.$.list.offsetWidth;
+        var rows = new Array();
+        var currentRowWidth = 0;
+        var currentRow = new Array();
+        items.forEach(function(item, idx) {
+          var clone = Object.assign({}, item);
+          // fallback to square dimensions if item doesn't have a size object in it's model
+          clone.size = clone.properties['picture:info'] || {width: 1, height: 1};
+          clone._view = {};
+          clone._view.index = this.items.indexOf(item);
+
+          // compute item width to fit a row with `rowHeight`
+          // new width = original width * rowHeight / original height
+          clone._view.width = clone.size.width * this.rowHeight / clone.size.height;
+          clone._view.height = this.rowHeight;
+
+          // if item fits, add it to current row
+          if (currentRowWidth + clone._view.width <= gridWidth) {
+            currentRow.push(clone);
+            currentRowWidth += clone._view.width;
+          } else {
+            // current item doesn't fit in current row
+            // fit items do width and push current row to rows
+            rows.push(this._fitItemsToWidth(currentRow, currentRowWidth, gridWidth));
+            // append current item (that didn't fit in current row) to a new row
+            currentRow = [clone];
+            currentRowWidth = clone._view.width;
+          }
+
+          if (idx === (items.length - 1)) {
+            // fit items do width and push current row to rows
+            rows.push(this._fitItemsToWidth(currentRow, currentRowWidth, gridWidth));
+          }
+        }.bind(this));
+        return rows;
+      },
+
+      /**
+       * Computes the ideal height of current row in order to increase the width of items and take advantage of the
+       * remaining width space.
+       */
+      _fitItemsToWidth: function(currentRow, currentRowWidth, gridWidth) {
+        var computedItemsWidths = currentRow.map(function (item) {
+          return item._view.width / currentRowWidth * gridWidth;
+        }.bind(this));
+        var computedHeight = gridWidth * this.rowHeight / currentRowWidth;
+        currentRow.forEach(function(item, idx) {
+          item._view.height = computedHeight;
+          item._view.width = computedItemsWidths[idx];
+        });
+        return currentRow;
+      },
+
+      /**
+       * Recomputes rows when the browser window is resized.
+       */
+      _resize: function() {
+        if (this.$.list._isVisible) {
```

Comment:
🤞 this private API does not change....

Reply:
why not just reproduce the logics `if (this.$.list.offsetWidth || this.$.list.offsetHeight) { ... }` here instead?